frontend/vre/utils/jsonld.model.js

```diff
@@ -1,11 +1,102 @@
 import {APICollection} from "./api.model";
 
 /**
- * Generic subclass of APICollection that parses incoming JSON-LD to a
- * list of subjects.
+ * The @graph property inside JSON-LD
+ * @typedef {Object} JSONLDGraph
+ */
+
+/**
+ * An individual subject definition inside JSON-LD
+ * @typedef {Object} JSONLDSubject
+ */
+
+export var JsonLdModel = Backbone.Model.extend({
+    idAttribute: '@id',
+
+})
+
+/**
+ * Generic subclass of APICollection that parses incoming compacted JSON-LD to an
+ * array of all subjects. The contents of subjects are left unchanged.
  */
 export var JsonLdCollection = APICollection.extend({
+    model: JsonLdModel,
     parse: function(response) {
+        if (!response.hasOwnProperty("@graph")) {
+            throw "Response has no @graph key, is this JSON-LD in compacted form?";
+        }
         return response["@graph"];
     }
 });
+
+/**
+ * Return a nested version of a given subject by adding to it the objects
+ * it refers to if they are found in the graph.
+ * The subject passed to this function as an argument is not changed.
+ * @param subjectsByID{Dictionary<JSONLDSubject>} - The full contents of the graph in JSON-LD
+ * @param subject{JSONLDSubject} - The subject including its predicates and objects to create a nested version of
+ * @param parentSubjectIDs{Array<String>} - For internal use of recursive function; leave undefined
+ * @returns {Object}
+ */
+export function nestSubject(subjectsByID, subject, parentSubjectIDs=undefined) {
+    if (typeof parentSubjectIDs === "undefined") {
+        parentSubjectIDs = [subject["@id"]];
+    } else {
+        parentSubjectIDs = Array.from(parentSubjectIDs);
+        parentSubjectIDs.push(subject["@id"]);
+    }
```

Comment:
```suggestion
export function nestSubject(subjectsByID, subject, parentSubjectIDs=[]) {
    parentSubjectIDs.push(subject["@id"]);
```

Reply:
Hmm, I wrote it this way to make sure the array is copied so that it is not altered by upcoming invocations of the function. Is that not a problem in JavaScript?

Reply:
Two things:

1. In Python, you must never default a function parameter to something mutable like a list, because that default value is created at function definition time rather than function invocation time. That problem does not exist in most other programming languages. In JavaScript, you can safely do `someParameter = []`.
2. You are right that the same array is reused when you don't create a new one by default. This means that you also have to do `parentSubjectIDs.pop()` at the end of the function. Sorry, did not pay attention to whether you were already doing that.

Reply:
Ah, thank you! I have applied your changes and added the `pop` call at the end of the function.